### PR TITLE
pythonPackages.asana: 0.7.0 -> 0.7.1

### PR DESCRIPTION
**Pull request by Ma27**
###### Motivation for this change

Also fixed the package build from
https://hydra.nixos.org/build/78900949.

`requests_oauthlib` seems to work fine at version 1.0 with `asana`, so
rather than creating our own override for `asana` it's fine to use 1.0
and patch `setup.py`.

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [x] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [x] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

File: pkgs/development/python-modules/asana/default.nix [hunk 1]
```diff
@@ -4,22 +4,23 @@
...
-    sha256 = "0786y3wxqxxhsb0kkpx4bfzif3dhvv3dmm6vnq58iyj94862kpxf";
+    sha256 = "0vmpy4j1n54gkkg0l8bhw0xf4yby5kqzxnsv07cjc2w38snj5vy1";
   };
 
   checkInputs = [ pytest responses ];
   propagatedBuildInputs = [ requests requests_oauthlib six ];
 
-  patchPhase = ''
+  postPatch = ''
     echo > requirements.txt
```
Comment (dotlambda) [inline pkgs/development/python-modules/asana/default.nix]: Is this line still needed?

**Comment by dotlambda** [comment]
What do you think of just applying https://github.com/Asana/python-asana/commit/80919ea8ae56fb8da5d90b0594009ac028068f73?

**Comment by Ma27** [comment]
@dotlambda isn't `>=0.8.0<2.0` equal to my constraint (`1.*`) in case of `requests_oauthlib`?

**Comment by dotlambda** [comment]
No, someone might actually want to use version 0.8.0.

**Comment by Ma27** [comment]
ack, fixed :+1:

**Pull request merged**